File: LC09.cpp
/*
Given an integer x, return true if x is a palindrome, and false otherwise.

(Even negative numbers are considered not palindromes)
 */


class Solution {
public:
    bool isPalindrome(int x) {
        long rev = 0, rem, p = x;

        if(x < 0)
            return false;

        while(x != 0){
            rem = x%10;
            rev = rev*10 + rem;
            x = x/10;
        }
        
        if(rev == p)
            return true;
        
        return false;
    }
};

File: LC09.py
'''
Given an integer x, return true if x is a palindrome, and false otherwise.
'''
class Solution:
    def isPalindrome(self, x: int) -> bool:
        rev = 0
        p = x
        
        if x < 0:
            return False
        
        while x != 0:
            rem = x%10
            rev = rev*10 + rem
            x //= 10
        
        return (rev == p)
